File: InfiniteViewUseCollectionV/ViewController.m
//
//  ViewController.m
//  InfiniteViewUseCollectionV
//
//  Created by April Lee on 2015/4/30.
//  Copyright (c) 2015年 april. All rights reserved.
//

#import "ViewController.h"
#import "ALInfiniteView.h"


@interface ViewController () <ALInfiniteViewDelegate>

@property (nonatomic) NSArray *colorItems;
@property (weak, nonatomic) IBOutlet UIView *colorView;

@property (strong, nonatomic) ALInfiniteView *infiniteView;

@end

@implementation ViewController

- (void)viewDidLoad {

    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self createInfiniteView];
    [self createColorItems];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - createInfiniteView

- (void)createInfiniteView {
    _infiniteView = [[ALInfiniteView alloc] initWithFrame:CGRectMake(0, 0, CGRectGetMaxX(_colorView.bounds), CGRectGetMaxY(_colorView.bounds)) owner:self];
    _infiniteView.delegate = self;
    [_infiniteView setAutoScrolled:YES];
    
    [_colorView addSubview:_infiniteView];
}


#pragma mark - setupInfiniteData

- (void)createColorItems {
    NSDictionary *grayColor = @{@"Name" : @"grayColor", @"Color" : [UIColor grayColor]};
    NSDictionary *purpleColor = @{@"Name" : @"purpleColor",@"Color" : [UIColor purpleColor]};
    NSDictionary *brownColor = @{@"Name" : @"brownColor" , @"Color" : [UIColor brownColor]};
    
    _colorItems = @[grayColor, purpleColor, brownColor];
    
    [_infiniteView createInfiniteItems:_colorItems];
}

#pragma mark - ALInfiniteViewDelegate

- (void)ALInfiniteViewDidSelectedItem:(ALInfiniteView *)infinteView selectedIndex:(NSInteger)selectedIndex {
    NSDictionary *selectedDic = _colorItems[selectedIndex];
    
    [self showAlertView:selectedDic[@"Name"]];
}

#pragma mark - AlertView Controller

- (void)showAlertView:(NSString *)message {
    UIAlertController *alertView = [UIAlertController alertControllerWithTitle:@"Color name" message:message preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {}];
    
    [alertView addAction:okAction];
    
    [self presentViewController:alertView animated:YES completion:nil];
}

@end

File: Swift/infiniteViewUseCollectionViewSwift/infiniteViewUseCollectionViewSwift/ViewController.swift
//
//  ViewController.swift
//  infiniteViewUseCollectionViewSwift
//
//  Created by April on 2017/7/18.
//  Copyright © 2017年 April. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var colorView: UIView!
    var colorItems : [[String: Any]] = [[:]]
    
    var infiniteView : ALInfiniteView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        createColorItems()
        
        infiniteView = ALInfiniteView(frame:colorView.frame)
        infiniteView?.delegate = self
        infiniteView?.setInfiniteViewItems(orginItems: colorItems)
        infiniteView?.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        
        colorView.addSubview(infiniteView!)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewDidAppear(_ animated: Bool) {
        if infiniteView != nil {
            infiniteView?.scrollToVisibleFirstItem()
        }
        
    }

    func createColorItems() {
        let grayColor : [String: Any] = ["Name": "Gray Color", "Color": UIColor.gray]
        let purpleColor : [String: Any] = ["Name": "Purple Color", "Color": UIColor.purple]
        let brownColor : [String: Any] = ["Name": "Brown Color", "Color": UIColor.brown]
        
        colorItems = [grayColor, purpleColor, brownColor]
    }
    
}

extension ViewController: ALInfiniteViewDelegate {
    func alInfiniteViewDidSelectedItem(infiniteView: ALInfiniteView, selectedIndex: Int) {
        
    }
}

